### PR TITLE
Events patch

### PR DESCRIPTION
**Pull request by wojtodzio**
Introducing new events view

### CODE REVIEW

File: _layouts/events.html [hunk 1]
```diff
@@ -2,23 +2,70 @@
...
+              {% when '7' %}Lip
+              {% when '8' %}Sie
+              {% when '9' %}Wrze
+              {% when '10' %}Paź
+              {% when '11' %}Lis
+              {% when '12' %}Gru
+              {% endcase %}
+            </p>
+          </a>
+		      <a href="{{ event.link }}" target="_blank">
```
Comment (ryrych) [inline _layouts/events.html]: broken formatting

File: _sass/base/_utility.scss [hunk 2]
```diff
@@ -32,4 +32,17 @@ a.button:hover {
...
+  content: ".";
+  display: block;
+  height: 0;
+  overflow: hidden;
+}
+.clearfix:after {
+    clear: both;
+}
+.clearfix {
+    zoom: 1; /* IE < 8 */
```
Comment (ryrych) [inline _sass/base/_utility.scss]: We can safely do without this ;)

File: _sass/includes/_events.scss [hunk 3]
```diff
@@ -1,54 +1,162 @@
+$mikstura_green: #1abc9c;
+@font-face {
+    font-family: 'Glyphicons Halflings';
+    src: url(../fonts/glyphicons-halflings-regular.woff);
+}
+@mixin text_overflow_lines ($lines) {
+  display: block;
+  display: -webkit-box;
```
Comment (ryrych) [inline _sass/includes/_events.scss]: what's the support for that?

File: _sass/includes/_events.scss [hunk 4]
```diff
@@ -1,54 +1,162 @@
...
+  -webkit-line-clamp: $lines;
+  -webkit-box-orient: vertical;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+.glyphicon {
+  position: relative;
```
Comment (ryrych) [inline _sass/includes/_events.scss]: Consider to sort properties alphabetically.

File: _sass/includes/_events.scss [hunk 5]
```diff
@@ -1,54 +1,162 @@
...
+  top: 1px;
```
Comment (ryrych) [inline _sass/includes/_events.scss]: this could be done with vertical-align

File: _sass/includes/_events.scss [hunk 6]
```diff
@@ -1,54 +1,162 @@
...
+  display: inline-block;
+  font-family: 'Glyphicons Halflings';
+  font-size: 1em;
+  line-height: 1;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+.glyphicon-comment:before {
+  content: "\e111";
+  margin-right: 0.2em;
```
Comment (ryrych) [inline _sass/includes/_events.scss]: Consider keeping all icons in a separate file. White space control / padding / margin should not be applied on a icon basis. If you really need to add some extra padding, add it to the specific element the icon is attached to.

File: _sass/includes/_events.scss [hunk 7]
```diff
@@ -1,54 +1,162 @@
...
-
-    .link-wrapper {
-      display: block;
-    }
-    .social-media-icon {
-      background-color: transparent;
-      border-radius: 50%;
-      color: black !important;
+    li.event-item-padded {
+      height: 23em;
```
Comment (ryrych) [inline _sass/includes/_events.scss:9]: Try to keep consistent vertical rhythm (line height). You'd be better off with `24em` as you could split it in half.

File: _sass/includes/_events.scss [hunk 8]
```diff
@@ -1,54 +1,162 @@
...
-
-      &:hover {
-        background-color: #1abc9c;
+      margin-bottom: 0.2em;
+      div.event-item{
+        position: relative;
+        border: 1px outset #ccc;
+        height: 100%;
+        width: 100%;
+			  p {
```
Comment (ryrych) [inline _sass/includes/_events.scss]: Please pay attention to formatting before pushing.

File: _sass/includes/_events.scss [hunk 9]
```diff
@@ -1,54 +1,162 @@
...
-      transition-function: ease-in;
-      transition-property: background-color;
-
-      &:hover {
-        background-color: #1abc9c;
+      margin-bottom: 0.2em;
+      div.event-item{
+        position: relative;
+        border: 1px outset #ccc;
+        height: 100%;
```
Comment (ryrych) [inline _sass/includes/_events.scss:15]: Isn't it default?

File: _sass/includes/_events.scss [hunk 10]
```diff
@@ -1,54 +1,162 @@
...
+        width: 100%;
+			  p {
+			    line-height: 1.15em;
+			    font-size: 0.75em;
```
Comment (ryrych) [inline _sass/includes/_events.scss]: Hard to know what's going on. Consider `$font-size-normal`, `$font-size-small`, etc.

File: _sass/includes/_events.scss [hunk 11]
```diff
@@ -1,54 +1,162 @@
...
+					height: 6em;
+					top: -20px;
+					text-align: center;
+					margin: 0.4em 0.7em;
+					z-index: 9;
+					left: -0.5em;
+					font-weight: 300;
+					color: #fff;
+					&:before{
+						font-family: 'Glyphicons Halflings';
```
Comment (ryrych) [inline _sass/includes/_events.scss]: Lot's of magic here :D

File: _sass/includes/_events.scss [hunk 12]
```diff
@@ -1,54 +1,162 @@
...
+		        width: 90%;
+		        border-top: 1px solid #ccc;
+		      }
+			    .event-tags {
+			      color: #666;
+			      font-size: 0.75em;
+			      text-align: center;
+			      padding: 0 0.5em;
+			      float: left;
+			      margin: 8px 0 0 16px;
```
Comment (ryrych) [inline _sass/includes/_events.scss]: You can use format  / reformat code in RubyMine.

File: _sass/includes/_events.scss (hunk 8 above)
Comment (wojtodzio) [inline _sass/includes/_events.scss]: Sorry about that, I've used Vim with :set tabstop=2 but without :set expandtab, and TextMate with opposite settings. That caused mess, which I was not even aware of. I'II fix it immediately.

File: _sass/base/_utility.scss [hunk 13]
```diff
@@ -1,3 +1,26 @@
...
+  src: url(../fonts/glyphicons-halflings-regular.woff);
+}
+.glyphicon {
+  display: inline-block;
+  font-family: 'Glyphicons Halflings';
+  font-size: 1em;
+  line-height: 1;
+  position: relative;
+  top: 1px;
+  -moz-osx-font-smoothing: grayscale;
```
Comment (Darex1991) [inline _sass/base/_utility.scss:12]: You can create mixin with it ;)

File: _sass/base/_utility.scss [hunk 14]
```diff
@@ -1,3 +1,26 @@
...
+  -webkit-font-smoothing: antialiased;
+}
+.glyphicon-comment:before {
```
Comment (Darex1991) [inline _sass/base/_utility.scss:15]: place it into another file with icons ;) ie icons., etc

File: _sass/base/_utility.scss [hunk 15]
```diff
@@ -1,3 +1,26 @@
...
+  content: "\e111";
+}
+.glyphicon-map-marker:before {
+  content: "\e062";
+}
+.glyphicon-calendar:before {
+  content: "\e109";
+}
 // mix-ins
```
Comment (Darex1991) [inline _sass/base/_utility.scss:24]: please place it into file "mixin" ;)

File: _sass/base/_utility.scss [hunk 16]
```diff
@@ -15,6 +38,23 @@
...
+  overflow: hidden;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: $lines;
+}
+%background-transition-hover {
+  transition-duration: .1s, .1s;
+  transition-property: background-color, color;
+  transition-timing-function: ease-in;
+  &:hover, &:active {
+    background-color: $mikstura_green;
```
Comment (Darex1991) [inline _sass/base/_utility.scss:54]: change name of this variable ;) you can check our project [link](https://github.com/Selleo/neutopia/tree/master/app/assets/stylesheets)

File: _sass/base/_utility.scss [hunk 17]
```diff
@@ -32,4 +72,17 @@ a.button:hover {
 // states
 .disabled {
   opacity: 0.7;
-}
\ No newline at end of file
+}
+.clearfix:before,
+.clearfix:after {
+  content: ".";
```
Comment (Darex1991) [inline _sass/base/_utility.scss:78]: Why "." ? ;)

File: _sass/base/_variables.scss [hunk 18]
```diff
@@ -28,3 +28,10 @@ $feature-image-size: cover; // options include "cover", "contain", "auto"
 // header description box
 $header-desc-background-color: #F98752; 
 $header-desc-text-color: #FFF;
+
+// from events-patch
+$mikstura_green: #1abc9c;
+$grey: #5C5C5C;
```
Comment (Darex1991) [inline _sass/base/_variables.scss:34]: you can add prefix, ie: '$color-grey'

File: _sass/includes/_events.scss [hunk 19]
```diff
@@ -1,53 +1,110 @@
...
+          color: $grey;
+          font-size: $font-size-small;
+          line-height: 1.15em;
+        }
+        .p-hov:hover, .p-hov:active{
+          color: #000;
+        }
+        .event-date{
+          color: #fff;
+          font-weight: 300;
```
Comment (Darex1991) [inline _sass/includes/_events.scss:28]: you can add font weight into variables ;)

File: _sass/includes/_events.scss [hunk 20]
```diff
@@ -1,53 +1,110 @@
...
+          margin: 0.4em 0.7em;
+          position: absolute;
+          text-align: center;
+          top: -20px;
+          width: 4em;
+          z-index: 9;
+          &:before{
+            color: $mikstura_green;
+            content: "\e044";
+            font-family: 'Glyphicons Halflings';
```
Comment (Darex1991) [inline _sass/includes/_events.scss:40]: here you can include mixin with icon ;) [link](https://github.com/Selleo/neutopia/blob/master/app/assets/stylesheets/shared/icons-new.sass)

**Pull request merged**